src/workflows/registry.rs

```diff
@@ -0,0 +1,32 @@
+use std::collections::HashMap;
+
+use super::Workflow;
+
+type WorkflowIdentifier = usize;
+
+pub trait WorkflowRegistry: Send + Sync {
+    fn register(&mut self, workflow: Workflow) -> WorkflowIdentifier;
+    fn load(&self, id: &WorkflowIdentifier) -> Option<Workflow>;
+}
+
+pub struct HashMapRegistry {
+    map: HashMap<usize, Workflow>
+}
+
+impl HashMapRegistry {
+    pub fn new() -> Self {
+        Self { map: HashMap::new() }
+    }
+}
+
+impl WorkflowRegistry for HashMapRegistry {
+    fn register(&mut self, workflow: Workflow) -> WorkflowIdentifier {
+        let id = self.map.len();
```

Comment:
Use hash of workflow? Now we can just increment ids to query the next workflow.

Reply:
done

Reply:
Not quite, see the other comment.